From afa2b690a8da8f05085357a8aea3d4b33334cf52 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:03:38 +0100
Subject: [PATCH 440/456] rtl8192du: open code flush_signals_thread()

flush_signals_thread() is only used once, so just open code it.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 core/rtw_cmd.c          | 4 +++-
 include/osdep_service.h | 6 ------
 2 files changed, 3 insertions(+), 7 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_cmd.c
+++ b/drivers/staging/rtl8192du/core/rtw_cmd.c
@@ -336,7 +336,9 @@ post_process:
 			}
 		}
 
-		flush_signals_thread();
+		if (signal_pending (current))
+			flush_signals(current);
+
 		goto _next;
 	}
 	pcmdpriv->cmdthd_running = false;
--- a/drivers/staging/rtl8192du/include/osdep_service.h
+++ b/drivers/staging/rtl8192du/include/osdep_service.h
@@ -158,12 +158,6 @@ u32	rtw_ms_to_systime(u32 ms);
 void	rtw_sleep_schedulable(int ms);
 void	rtw_usleep_os(int us);
 
-static inline void flush_signals_thread(void)
-{
-	if (signal_pending (current))
-		flush_signals(current);
-}
-
 static inline int res_to_status(int res)
 {
 	return res;
